File: app/src/main/java/com/lilaclab/dadimadu/database/entity/Pelanggan.java
package com.lilaclab.dadimadu.database.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName="pelanggan")
public class Pelanggan {
    @PrimaryKey(autoGenerate=true)
    @ColumnInfo(name="id_pelanggan")
    public int idPelanggan;
    @ColumnInfo(name="nama_pelanggan")
    public String namaPelanggan;
    @ColumnInfo(name="no_hp")
    public String noHp;
    public String alamat;

    public Pelanggan() {
    }

    @Ignore
    public Pelanggan(String namaPelanggan, String noHp, String alamat) {
        this.namaPelanggan = namaPelanggan;
        this.noHp = noHp;
        this.alamat = alamat;
    }
}

File: app/Models/Pelanggan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Pelanggan extends Model
{
    protected $table      = 'pelanggan';
    protected $primaryKey = 'id_pelanggan';
    public    $timestamps = false;

    const CREATED_AT = 'create_at';
    const UPDATED_AT = null;

    protected $fillable = ['nama_pelanggan', 'no_hp', 'alamat'];

    public function transaksi()
    {
        return $this->hasMany(Transaksi::class, 'id_pelanggan', 'id_pelanggan');
    }
}
